# Construction of pYPK0_FBA1_ScTAL1_PDC1 (pMEC1130)


[pYPKa_Z_FBA1](pYPKa_Z_FBA1.ipynb)

[pYPKa_E_PDC1](pYPKa_E_PDC1.ipynb)

In [26]:
import pydna
from pydna import ipynb_importer

In [27]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [28]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [29]:
pYPK0 = pydna.read("pYPK0.gb")

In [30]:
promoter_clone = pYPKa_Z_FBA1 = pydna.read("pYPKa_Z_FBA1.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001182243.1) record

In [31]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169869 width=100% height=250></iframe>')

In [32]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001182243.1")

In [33]:
gene_template.list_features()

+----------+-----------+-------+-------+--------+--------------+--------+------+
| Feature# | Direction | Start |  End  | Length | id           | type   | orf? |
+----------+-----------+-------+-------+--------+--------------+--------+------+
| 0        |    -->    |   0   |  1008 |   1008 | <unknown id> | source | yes  |
| 1        |    -->    |   <0  | >1008 |   1008 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   |  1008 |   1008 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+-------+--------+--------------+--------+------+

In [34]:
orf = gene_template.extract_feature(2)

In [35]:
orf.isorf()

True

In [36]:
print len(orf)
orf.seguid()

1008


GgJ9s-lK11rrZmXZDHjE3AjiVrs

In [37]:
print str(orf.seq.translate())

MSEPAQKKQKVANNSLEQLKASGTVVVADTGDFGSIAKFQPQDSTTNPSLILAAAKQPTYAKLIDVAVEYGKKHGKTTEEQVENAVDRLLVEFGKEILKIVPGRVSTEVDARLSFDTQATIEKARHIIKLFEQEGVSKERVLIKIASTWEGIQAAKELEEKDGIHCNLTLLFSFVQAVACAEAQVTLISPFVGRILDWYKSSTGKDYKGEADPGVISVKKIYNYYKKYGYKTIVMGASFRSTDEIKNLAGVDYLTISPALLDKLMNSTEPFPRVLDPVSAKKEAGDKISYISDESKFRFDLNEDAMATEKLSEGIRKFSADIVTLFDLIEKKVTA*


In [38]:
pf, pr = pydna.parse('''>TAL1_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA
                        >TAL1_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAAGCGGTAACTTTCTTTTCAATCA''')

In [39]:
g = pydna.pcr( pf, pr, orf)

In [40]:
g.figure()

                                        5ATGTCTGAACCAGCTCAAAAGAA...TGATTGAAAAGAAAGTTACCGCTTAA3
                                                                   |||||||||||||||||||||||||| tm 52.1 (dbd) 65.6
                                                                  3ACTAACTTTTCTTTCAATGGCGAATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGTCTGAACCAGCTCAAAAGAA3
                                         ||||||||||||||||||||||| tm 52.6 (dbd) 65.3
                                        3TACAGACTTGGTCGAGTTTTCTT...ACTAACTTTTCTTTCAATGGCGAATT5

In [41]:
terminator_clone = pYPKa_E_PDC1 = pydna.read("pYPKa_E_PDC1.gb")

In [42]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [43]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [44]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Dseqrecord(-792), Dseqrecord(-1086), Dseqrecord(-1061))

In [45]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [46]:
asm

Assembly:
Sequences........................: [5681] [792] [1086] [1061]
Sequences with shared homologies.: [5681] [792] [1061] [1086]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [8374]
Linear products..................: [8472] [8435] [8424] [8411] [7730] [7411] [7375] [6681] [6375] [2852] [2110] [1828] [98] [61] [50] [37]

In [47]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|792bp_PCR_prod|38
|                          \/
|                          /\
|                          38|1086bp_PCR_prod|35
|                                             \/
|                                             /\
|                                             35|1061bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [48]:
pYPK0_FBA1_ScTAL1_PDC1 = candidate.synced(pYPK0)

The pYPK0_FBA1_ScTAL1_PDC1 has cseguid ```z56XvyucVzpo2iNPYy5TbXbxWlk``` and 8374 bp.

In [49]:
print len(pYPK0_FBA1_ScTAL1_PDC1)
pYPK0_FBA1_ScTAL1_PDC1.cseguid()

8374


z56XvyucVzpo2iNPYy5TbXbxWlk

In [50]:
pYPK0_FBA1_ScTAL1_PDC1.name="pMEC1130"
#pYPK0_FBA1_ScTAL1_PDC1.description = "pYPK0_FBA1_ScTAL1_PDC1tp"
pYPK0_FBA1_ScTAL1_PDC1.stamp()

cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk_2015-07-15T12:32:52.717518

In [51]:
pYPK0_FBA1_ScTAL1_PDC1.write("pYPK0_FBA1_ScTAL1_PDC1.gb")

###[pYPK0_FBA1_ScTAL1_PDC1](pYPK0_FBA1_ScTAL1_PDC1.gb)

##Download sequence

[pYPK0_FBA1_ScTAL1_PDC1](pYPK0_FBA1_ScTAL1_PDC1.gb)

In [52]:
import pydna

In [53]:
reloaded = pydna.read("pYPK0_FBA1_ScTAL1_PDC1.gb")

In [54]:
reloaded.cseguid()

z56XvyucVzpo2iNPYy5TbXbxWlk

In [55]:
reloaded.description

'cSEGUID_z56XvyucVzpo2iNPYy5TbXbxWlk_2015-07-15T12:32:52.717518'